packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -265,6 +265,273 @@ void VideoPlayer::SetPlaybackSpeed(double speed) {
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_get_state failed: %s",
+              get_error_message(ret));
+    return {};
+  }
+  if (state == PLAYER_STATE_NONE || state == PLAYER_STATE_IDLE) {
+    LOG_ERROR("[VideoPlayer] Player not ready.");
+    return {};
+  }
+
+  void *player_lib_handle = dlopen("libcapi-media-player.so.0", RTLD_LAZY);
+  if (!player_lib_handle) {
+    LOG_ERROR("[VideoPlayer] dlopen failed: %s", dlerror());
+    return {};
+  }
+
+  FuncPlayerGetTrackCountV2 player_get_track_count_v2 =
+      reinterpret_cast<FuncPlayerGetTrackCountV2>(
+          dlsym(player_lib_handle, "player_get_track_count_v2"));
+  FuncPlayerGetSubtitleTrackInfoV2 player_get_subtitle_track_info_v2 =
+      reinterpret_cast<FuncPlayerGetSubtitleTrackInfoV2>(
+          dlsym(player_lib_handle, "player_get_subtitle_track_info_v2"));
+
+  if (!player_get_track_count_v2 || !player_get_subtitle_track_info_v2) {
+    LOG_ERROR("[VideoPlayer] Symbol not found: %s", dlerror());
+    dlclose(player_lib_handle);
+    return {};
+  }
+
+  int subtitle_count = 0;
+  ret = player_get_track_count_v2(player_, PLAYER_STREAM_TYPE_TEXT,
+                                  &subtitle_count);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("[VideoPlayer] player_get_subtitle_track_count failed: %s",
+              get_error_message(ret));
+    return {};
+  }
+  LOG_INFO("[VideoPlayer] subtitle_count: %d", subtitle_count);
+
+  player_subtitle_track_info_v2 *sub_track_info = NULL;
+  flutter::EncodableList trackSelections;
+
+  if (subtitle_count > 0) {
+    for (int sub_index = 0; sub_index < subtitle_count; sub_index++) {
+      flutter::EncodableMap trackSelection = {};
+      ret = player_get_subtitle_track_info_v2(player_, sub_index,
+                                              &sub_track_info);
+      if (ret != PLAYER_ERROR_NONE) {
+        LOG_ERROR("[VideoPlayer] player_get_subtitle_track_info_v2 failed: %s",
+                  get_error_message(ret));
+        return {};
+      }
+      LOG_INFO(
+          "[VideoPlayer] subtitle track info: language[%s], "
+          "subtitle_type[%d]",
+          sub_track_info->language, sub_track_info->subtitle_type);
+
+      trackSelection.insert({flutter::EncodableValue("trackType"),
+                             flutter::EncodableValue(PLAYER_STREAM_TYPE_TEXT)});
+      trackSelection.insert({flutter::EncodableValue("trackId"),
+                             flutter::EncodableValue(sub_index)});
+      trackSelection.insert(
+          {flutter::EncodableValue("language"),
+           flutter::EncodableValue(std::string(sub_track_info->language))});
+      trackSelection.insert(
+          {flutter::EncodableValue("subtitleType"),
+           flutter::EncodableValue(sub_track_info->subtitle_type)});
+
+      trackSelections.push_back(flutter::EncodableValue(trackSelection));
+    }
+  }
+  dlclose(player_lib_handle);
+  return trackSelections;
+}
+
```

Comment:
we can extract  a new function for common operation to get track info,
we can move all duplicate code to this function.